example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js

```diff
@@ -0,0 +1,279 @@
+
+    expect(getByText("##")).toBeInTheDocument();
+  });
+
+  it("should render h3 markdown when clicking the h3 button", async () => {
+    const { container, getByText } = render(
+      <ThemeProvider theme={lightTheme}>
+        <Wysiwyg label={"hello world"} placeholder={""} onChange={jest.fn()} />
+      </ThemeProvider>
+    );
+
+    await waitFor(() => container.querySelector(".CodeMirror-cursor"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h3"));
+
+    expect(getByText("###")).toBeInTheDocument();
+  });
+
+  it("should render h4 markdown when clicking the h4 button", async () => {
+    const { container, getByText } = render(
+      <ThemeProvider theme={lightTheme}>
+        <Wysiwyg label={"hello world"} placeholder={""} onChange={jest.fn()} />
+      </ThemeProvider>
+    );
+
+    await waitFor(() => container.querySelector(".CodeMirror-cursor"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h4"));
+
+    expect(getByText("####")).toBeInTheDocument();
+  });
+
+  it("should render h5 markdown when clicking the h5 button", async () => {
+    const { container, getByText } = render(
+      <ThemeProvider theme={lightTheme}>
+        <Wysiwyg label={"hello world"} placeholder={""} onChange={jest.fn()} />
+      </ThemeProvider>
+    );
+
+    await waitFor(() => container.querySelector(".CodeMirror-cursor"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h5"));
+
+    expect(getByText("#####")).toBeInTheDocument();
+  });
+
+  it("should render h6 markdown when clicking the h6 button", async () => {
+    const { container, getByText } = render(
+      <ThemeProvider theme={lightTheme}>
+        <Wysiwyg label={"hello world"} placeholder={""} onChange={jest.fn()} />
+      </ThemeProvider>
+    );
+
+    await waitFor(() => container.querySelector(".CodeMirror-cursor"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h6"));
+
+    expect(getByText("######")).toBeInTheDocument();
+  });
+
+  it("should render h1 markdown when clicking the h4 button then clicking on the h1 button", async () => {
+    const { container, queryByText, getByText } = render(
+      <ThemeProvider theme={lightTheme}>
+        <Wysiwyg label={"hello world"} placeholder={""} onChange={jest.fn()} />
+      </ThemeProvider>
+    );
+
+    await waitFor(() => container.querySelector(".CodeMirror-cursor"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h1"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h4"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h1"));
+
+    expect(queryByText("####")).not.toBeInTheDocument();
+    expect(getByText("#")).toBeInTheDocument();
+  });
+});
```

Comment:
Super cool tests 😄 . If you want to do something before each test like the `render(...)`, you can call the `beforeEach()` function.

```js
describe("Wysiwyg", () => {
    let renderedContainer;
    let getSomethingByText;
    let querySomethingByText;

    beforeEach(() => {
        const { container, getByText, queryByText } = render(
            <ThemeProvider theme={lightTheme}>
               <Wysiwyg label={"hello world"} placeholder={""} onChange={jest.fn()} />
            </ThemeProvider>
         );
         renderedContainer = container;
         getContainerByText = getByText;
         containerQueryByText = queryByText;
    });

    // In your tests
    it("should ...",  () => {
         await waitFor(() => renderedContainer.querySelector(".CodeMirror-cursor"));
         expect(getContainerByText("hello world")).toBeInTheDocument();
         // ...;
    });
});
```

Reply:
nice tip thank you, will try it!